### PR TITLE
Repo2docker dask fixes

### LINKED ISSUE #47: onbuild environment.yml going to /srv/conda instead of /srv/conda/envs/notebook
Our onbuild images are now installing extra packages from environment.yml to the base conda environment (/srv/conda/) instead of the default environment (/srv/conda/envs/notebook). 

Seems related to a recent change in repo2docker
https://github.com/jupyter/repo2docker/pull/651 

I'm hoping this is a simple fix in setting an environment variable (NB_PYTHON_PREFIX ?), but @yuvipanda or @jhamman will know!

### PR DESCRIPTION
**Pull request by scottyhq**
hopefully fixes #47 and #48

### CODE REVIEW

File: base-notebook/binder/environment.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ dependencies:
   - dask-kubernetes>=0.8.0
   - nbserverproxy
   - ipywidgets
+  - graphviz
```
Comment (jhamman) [inline base-notebook/binder/environment.yml]: Happy to try this again but I have a seem to remember this not working in a recent test. We'll see.

File: onbuild/r2d_overlay.py [hunk 2]
```diff
@@ -46,10 +47,10 @@ def apply_environment():
 def apply_requirements():
     req_path = binder_path('requirements.txt')
     env_path = binder_path('environment.yml')
-
+    NB_PYTHON_PREFIX=os.environ['NB_PYTHON_PREFIX']
```
Comment (jhamman) [inline onbuild/r2d_overlay.py]: nit: (here and above), can you add a space around the `=`?

Comment (scottyhq) [reply onbuild/r2d_overlay.py]: ok. added tests, including checking currently pinned versions - could be more sophisticated, but i think is good for now

**Comment by jhamman** [comment]
@scottyhq - you may want to check out this: https://github.com/jupyter/repo2docker/pull/690

**Pull request merged**